### DISCUSSION TITLE
Feature Request: AI Alternative - Anthropic / Self-Hosted

### DISCUSSION THREAD

**Discussion by PrestoMac** [AppFlowy-IO/AppFlowy#3851] [Ideas]
**Hey Dev Team!**
First off, huge kudos for creating AppFlowy! It's amazing to see the capabilities that are possible with an open-source alternative to Notion.

I wanted to put in a request for the ability to use alternate AI models besides just OpenAI, specifically Anthropic's Claude or a self-hosted model.

Some potential benefits of allowing different models:
- Gives users more choice and control over their data and interactions
- Allows for customization to suit different needs or preferences
- Enables using models that may be more affordable, private, or aligned with certain values
- Fosters competition and innovation between AI providers
- Future-proofs AppFlowy as new models emerge

I imagine there could be some complexities to implement this, but wanted to get your thoughts given the momentum and interest around AI assistants nowadays. Even just supporting Claude as an option alongside GPT would be awesome!

Let me know if any additional context would help or if you have any other feedback. Loving AppFlowy so far and excited to see where you take it next!

**Comment by AnsahMohammad** [comment]
Expanding on that idea, an excellent addition would be the capability to integrate various open-source AI models. Many of these models operate without the necessity for tokens, enabling users to harness the power of AI effortlessly within their workflow
Advantages : 
- Incorporate AI directly into their workflow 
- No need to sign in and generate a token

Disadvantages : 
- Some of the models will be less accurate ( Solution: Select only high-quality models)
- May confuse the user with a lot of options ( Solution: Add the options to switch to various AI Models in the settings and not in the main selection area)

**Reply by Fau57** [reply]
Llama.cpp or llama.rn integration for pc - mobile should work, imho

**Comment by richardstevenhack** [comment]
I agree. "Your data is yours forever" is not true if using OpenAI, especially since they just appointed a former NSA director to the board. I trust only local AIs for my private notes. It's fine having the ability to use a cloud AI but it should not be limited to that. I can use cloud and local AIs with Obsidian.

**Comment by annieappflowy** [comment]
Thanks, everyone on this thread. Our vision for AppFlowy AI is accessible, collaborative, and contextual.
Accessible:
- Everyone can run AI on their machine - ultimate privacy, no GPU or Internet required
- Access to advanced models - GPT4O, Claude 3, and more
- Please bear with us as it takes time to realize our vision. We will be enabling Claude 3 soon, and then local AI will come next.


![image](https://github.com/AppFlowy-IO/AppFlowy/assets/12026239/f63ca4de-6b43-47d9-b985-000cbcb95468)
https://github.com/AppFlowy-IO/AppFlowy/issues/4529

**Reply by nnWhisperer** [reply]
Where has openai code gone now? There isn't any openai code in the codebase anymore ?!

**Reply by sbolten** [reply]
what about adding gemini models? they are powerful and have a huge context window (the flash-lite for example)
